dev-tools/omdb/src/bin/omdb/db/saga.rs

```diff
@@ -437,3 +434,215 @@ async fn get_all_sagas_in_state(
+            Self::NoDnsServiceRecord(id) => (
+                format!(
+                    "Cannot proceed: no SRV record for Nexus with id {id}, \
+                    so cannot verify that it is not still running!"
+                ),
+                Severity::Error,
+            ),
+            Self::DnsIpv6LookupFailed { target, error } => (
+                format!(
+                    "Cannot proceed: failed to obtain Nexus IPv6 address for \
+                    {target}: {error}"
+                ),
+                Severity::Error,
+            ),
+            Self::NoAddressFromDns(target) => (
+                format!(
+                    "Cannot proceed: no AAAA record for Nexus with id {target},
+                    so cannot verify that it is not still running!"
+                ),
+                Severity::Error,
+            ),
+            Self::SecAnsweredPing(id) => (
+                format!(
+                    "Cannot proceed: Nexus with id matching current SEC \
+                    responded ok to a ping, meaning it is still running. \
+                    Abandoning or injecting errors into a running saga is not \
+                    safe. Please ensure the Nexus with id {id} is stopped \
+                    before proceeding."
+                ),
+                Severity::Error,
+            ),
+            Self::SecPingError { sec_id: id, .. } => (
+                format!(
+                    "Cannot proceed: Nexus with id matching current SEC \
+                    responded with an error to a ping, meaning it is still \
+                    running. Abandoning or injecting errors into a running \
+                    saga is not safe. Please ensure the Nexus with id {id} is \
+                    stopped before proceeding."
+                ),
+                Severity::Error,
+            ),
+            Self::SecAppearsInactive { sec_id, observed_error } => (
+                format!(
+                    "saw {observed_error} when trying to ping Nexus with id \
+                    {sec_id}. Proceed?"
+                ),
+                Severity::Info,
+            ),
+        };
+
+        match severity {
+            Severity::Info => println!("{msg}"),
+            Severity::Warning => {
+                if should_print_color {
+                    println!("{}", msg.yellow().bold());
+                } else {
+                    println!("{msg}");
+                }
+            }
+            Severity::Error => {
+                if should_print_color {
+                    println!("{}", msg.red().bold());
+                } else {
+                    println!("{msg}");
+                }
+            }
+        }
+    }
+
+    /// If this status indicates that the relevant SEC might be active, returns
+    /// `Err`. If the relevant SEC is thought to be inactive, or the saga used
+    /// to produce this status had no SEC, returns `Ok`.
+    fn as_result(self) -> anyhow::Result<()> {
```

Comment:
nits: should probably be `into_result` since this consumes `self`. Also, the doc comment could be a little clearer about what's returned here: it's `Ok` if there's sufficient evidence to be confident that this saga doesn't belong to an active SEC and `Err` otherwise.